documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
+- [ ] **M2005**: A longrunning operation *MUST* have a terminal success status code. At least 
+   - 200 or 201 for PUT/PATCH
+   - 200, 201 or 204  or all of the aforementioned for POST
+   - 204 or 200 or both for delete. 
+   This will indicate the code generator, how to deserialize the final success response after the polling is done. If the response has a response body then it *MUST* be modeled in the terminal status code.
 
-&#x25a2; *M2007*: Error definitions MUST be abstracted, modeled and reused instead of introducing a plethora of error definitions that look the same with different names.
+- [ ] **M2005**: Required parameters *MUST* be accurately labeled as `"required": true` in Swagger.
 
-&#x25a2; *M2008*: A Swagger spec MUST NOT define operations or properties or parameters for functionalities that MAY work in the future.
+- [ ] **M2006**: Required properties of a model definition *MUST* be accurately labeled as `"required": [ "propertyName1" ]` in Swagger.
 
-For example, `OutBoundNATRule` is a model that exists in a the networking Swagger but it is not supported by the Networking RP.
+- [ ] **M2007**: Mutability of properties *MUST* be marked up using [`x-ms-mutability` extension](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-mutability).
 
-&#x25a2; *M2009*: A collection model in Swagger MUST support adding more than one element to the collection.
+- [ ] **M2008**: Error definitions *MUST* be abstracted, modeled and reused.
 
-&#x25a2; *M2010*: Operations that are exposed through private previews MUST not be described in Swaggers in the [public GitHub repository](https://github.com/Azure/azure-rest-api-specs).
+- [ ] **M2009**: A Swagger spec *MUST NOT* define operations or properties or parameters for functionalities that are not currently supported by the service in the given api-version. The primary goal of the spec is that it *MUST* correctly and completely represent the underlying REST API.
 
+- [ ] **M2010**: A collection model in Swagger *MUST* support adding more than one element to the collection. TODO: Ask for clarity.
 
-## Operational - Swagger Checklist ##
+- [ ] **M2011**: Operations that are exposed through private previews *MUST NOT* be described in the [public GitHub repository](https://github.com/Azure/azure-rest-api-specs). They *MUST* be described in the [private GitHub repository](https://github.com/Azure/azure-rest-api-specs-pr) instead.
+
+- [ ] **M2012**: The value of the ["x-ms-client-name"](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-client-name) and ["x-ms-discriminator-value"](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-discriminator-value) *MUST NOT* be the same as the model property or the model name, otherwise the purpose is defeated.
+
+- [ ] **M2013**: For specifications describing the Management Plane of ARM, "subscriptionId" & "api-version" *MUST* always be defined in the global parameters section.
```

Comment:
Ideally, these would be common definitions. But please note that some APIs take two API versions (i.e. some of the resources APIs), and this guidance does not apply there.